---
layout:     post
title:      使用FastaQC和Cutadapt预处理miRNA-seq
subtitle:   How to preprocess miRNA-seq with FastQC and cutadapt
date:       2020-02-09
author:     Xuan
header-img: img/post-bg-miRNA.jpg
catalog: true
tags:
    - miRNA-seq
---

# How to preprocess miRNA-seq with FastQC and cutadapt

Aim: conduct research on miRNA 

Data: SRR5233942 .fastq directly from EBI 

The data also can be downloaded from NCBI-SRA, followed by type transform (from .SRA to .fasta/fastq); or Download with SRAtools.

[Reference](https://www.jianshu.com/p/cf0a7b937413)

## FastQC / Fastp
Deal with miRNA sequence, Fastp tool can not detect adapters, which can be detected by FastQC. Here is an example.

Tips: 1.If you run FastQC with Linux command, run with file names; otherwise it will print errors

![Image](/1.png)

## Cutadapt
- Search for standard adapter sequence, [Here](http://www.eurofinsgenomics.eu/media/1610545/illumina-adapter-sequences.pdf) is Illumina Adapter Sequence.

>"TruSeq Small RNA"

>1.RNA 5’ Adapter (RA5): 
5’ GUUCAGAGUUCUACAGUCCGACGAUC

>2.**RNA 3’ Adapter (RA3)**: 
5’ TGGAATTCTCGGGTGCCAAGG

>3.Stop Oligo (STP): 
5’ GAAUUCCACCACGUUCCCGUGG

>4.RNA RT Primer (RTP): 
5’ GCCTTGGCACCCGAGAATTCCA

>5.RNA PCR Primer (RP1): 
5’ AATGATACGGCGACCACCGAGATCTACACGTTCAGAGTTCTACAGTCCGA

- Run [Cutadapt](https://cutadapt.readthedocs.io/en/stable/guide.html) with -a parameter : (cutadapt -a TGGAATTCTCGGGTGCCAAGG -o SRR5233942_cut.fastq SRR5233942.fastq) **Recommand!**

>Tips: If there is multiple adapters, user can provide a Fasta file with "-g file:barcodes.fasta". 
>**While** in my test, with -g parameter, confused results are obtained.

- After adapter cut, all reads are received. so need to Fliter:

>**-m LENGTH**: Discard processed reads that are shorter than LENGTH.

>-M LENGTH: Discard processed reads that are larger than LENGTH.

>**--untrimmed-output FILE**: Write all reads without adapters to FILE (in FASTA/FASTQ format) instead of writing them to the regular output file.

###cutadapt shell script
```
cutadapt -a TGGAATTCTCGGGTGCCAAGG --untrimmed-output untrimmed.out -o out.fastq SRR5233942.fastq 
cutadapt -a TATGGAGTTCTACAGTCCGACGATCTA --untrimmed-output untrimmed.out -o out-2.fastq out.fastq
```

## Case Study

A preprocess script for a raw miRNA seq (SRR5233942)

```
fastqc SRR5233942.fastq
# 从Adapter Content 和 overrepresented sequences 中找到接头序列
# cutadapt 切除接头 和 未被处理序列
cutadapt -a TGGAATTCTCGGGTGCCAAGG --untrimmed-output untrimmed.out -o out.fastq SRR5233942.fastq 
cutadapt -a TATGGAGTTCTACAGTCCGACGATCTA --untrimmed-output untrimmed.out -o out-2.fastq out.fastq
# 按miRNA数据库统计的长度范围取16-35长的seq
cutadapt -m 16 -M 35 -o out_len.fastq out_cut.fastq
```

```markdown
**Bold** and _Italic_ and `Code` text

```
For more details see [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/).



### Support or Contact

xzhangxmu@gmail.com

