---
layout:     post
title:      综述k-mer counting方法
subtitle:   A review of k-mer counting methods
date:       2020-02-14
author:     Xuan
header-img: img/post-bg-miRNA.jpg
catalog: true
tags:
    - k-mer
---

# A review paper : "A benchmark study of k-mer counting methods for high-throughput sequencing"


## Catagories of k-mer counting methods

![Ontology of k-mer counting approaches](/img/post-ct-cata.png)

- Using the sorting approach
> sorting all k-mers extracted from each read; Thus k-mer frequencies can be easily counted because, after sorting, repeating k-mers lay at adjacent positions in the sorted list.  
**sorting and compaction (SAC) -based algorithm ??**    
aTurtle[used] / cTurtle  
**Pattern Block Bloom Filter** is a cache-friendly variant of the BLoom Filter, which has a very small cache miss ratio.  

- Using a hash table
> k-mer are stored as keys and their counts are stored as values.   
e.g. Jellyfish, **lock-free hash table** and **Compare-and-swap(CAS) assembly instruction** 
Jellyfish2, use an additional Bloom filter-based mode to remove all singleton k-mers. 
KAT

- Using a Bloom Filter and its Variants and counting quotient filter
> majority of any genomic dataset is made up of single-frequency k-mers, which mainly attributed to sequencing errors. 
A bloom filter is a probabilistic data structure used for dynamic membership query lookup. 
BFCounter. 
Squeaker: uses a counting filter data structure(**counting quotient filter[CQF]**)

- Using the concept of super k-mer: minimizers and signatures
> MSPKmerCounter is the 1st tool to implement **MSP(Minimum Substring Partitioning)** for k-mer counting.  
KMC3. 


## Related statistics

- 计算基因组🧬的大小
> 把每个k-mer看作基因组中的一个碱基，通过k-mer分布 -> k-mer总数（T）和每个碱基平均出现的次数（u).  
基因组大小 = T/u 

- 重复序列
> ?? 理论认为， 但拷贝序列k-mer，出现在主峰1.6倍以后的概率非常低；认为该区间为重复k-mer.   
重复序列长度 = 重复k-mer序列总数/ u

- 杂合率估算


