---
layout:     post
title:      SRA fastq Data Download
subtitle:   how to get raw sequence datasets in batches
date:       2021-02-20
author:     Xuan
header-img: img/post-bg-1.jpg
catalog: true
tags:
    - SRR 
    - sequence
---

# How to download fastq data?
## 1.USE Official recommended software SRAtool Kit

### Step1 Download [NCBI SRA Toolkit](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=software)

```
# Download step
wget [choosed link]
tar zxvf *.tar.gz
cd *
pwd

# Install & add PATH 

#. Way 1
echo 'export export PATH=$PATH:YOUR_PATH/sratoolkit.2.9.6-ubuntu64/bin' >> ~/.bash_profile
source ~/.bash_profile

#. Way 2
vim ~/.bash_profile
source ~/.bash_profile
```

? 修改～/.bashrc V.S. ～/bash_profile 的区别

### Step2 Use prefetch command obtain .sra files

**-O** --output-directory <DIRECTORY>  save files to DIRECTORY

```
prefetch SRR13338290 -O <DIRECTORY>

```


### Step3 Use fastq-dump command obtain .fastq files

```
prefetch SRR13338290

```

#### [Frequently Used Tools (in sratool Kit)](https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc):

fastq-dump: Convert SRA data into fastq format

prefetch: Allows command-line downloading of SRA, dbGaP, and ADSP data

sam-dump: Convert SRA data to sam format

sra-pileup: Generate pileup statistics on aligned SRA data

vdb-config: Display and modify VDB configuration information

vdb-decrypt: Decrypt non-SRA dbGaP data ("phenotype data")

Additional Tools:
abi-dump: Convert SRA data into ABI format (csfasta / qual)

illumina-dump: Convert SRA data into Illumina native formats (qseq, etc.)

sff-dump: Convert SRA data to sff format

sra-stat: Generate statistics about SRA data (quality distribution, etc.)

vdb-dump: Output the native VDB format of SRA data.

vdb-encrypt: Encrypt non-SRA dbGaP data ("phenotype data")

vdb-validate: Validate the integrity of downloaded SRA data
    

## Download in Batches

Go to [SRA Run Selector](https://www.ncbi.nlm.nih.gov/Traces/study/?) 
For example [SRP299920](https://www.ncbi.nlm.nih.gov/Traces/study/?query_key=12&WebEnv=MCID_6008c6d9ad7ec67b4272439c&o=acc_s%3Aa)
 
SRP299920: includes 20 SRR* files, you can selected SRR flies you interested, then choose "Accession List" to obtain all SRR files list (in SRR_Acc_List.txt). then you can use **prefetch command with file option** to download all files.

```
prefetch --option-file SRR_Acc_List.txt
```

```
SRR_Acc_List.txt
SRR13338271
SRR13338272
SRR13338273
SRR13338274
SRR13338275
SRR13338276
SRR13338277
SRR13338278
SRR13338279
SRR13338280
SRR13338281
SRR13338282
SRR13338283
SRR13338284
SRR13338285
SRR13338286
SRR13338287
SRR13338288
SRR13338289
SRR13338290
```

After Downloaded, data(.sra files) are stored in **～/ncbi/public/sra**

## 2. USE SRA Explorer: you can download fastq file directly

[SRA Explorer](https://sra-explorer.info/#), This tool aims to make datasets within the Sequence Read Archive more accessible.

### Step1 Search Accession number 

### Step2 Selected Datasets and way for downloading

optional ways:

- Raw FastQ Download URLs
- Bash script for downloading FastQ files
- Aspera commands for downloading FastQ files
- Cluster Flow FastQ download file (nice filenames)
- bcbio project file for FastQ downloads (nice filenames) 

### Step3 Copy command into a script and run by "nohup script.sh &"

### Step4 unzip SRR13338275.fastq.gz

```
gzip 
```

# Reference

1. 推荐参考Aspera快速下载部分. [SRA Toolkit - prefetch 快速下载NCBI SRA数据](https://www.jianshu.com/p/d1abdced8bcd)
2. [文件下载流程](https://www.yuque.com/biotrainee/wes/rl9v6b)

